File: external/cpp-impl/src/hgoe/nodes/term_node.cpp
//
// Created by jldevezas on 3/8/18.
//

#include <hgoe/nodes/term_node.h>

BOOST_CLASS_EXPORT_IMPLEMENT(TermNode)

TermNode::TermNode() = default;

TermNode::TermNode(std::string name) : Node(boost::move(name)) {}

Node::Label TermNode::label() const {
    return Label::TERM;
}

void TermNode::print(std::ostream &os) const {
    os << "TermNode ";
    Node::print(os);
}

File: external/java-impl/src/main/java/armyant/hgoe/nodes/TermNode.java
package armyant.hgoe.nodes;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import armyant.hgoe.RankableAtom;

/**
 * Created by jldevezas on 2017-10-24.
 */
public class TermNode extends Node implements RankableAtom {
    private static final Logger logger = LoggerFactory.getLogger(TermNode.class);

    public TermNode() {
    }

    public TermNode(String name) {
        super(name);
    }

    @Override
    public void setID(String id) {
        logger.warn("Ignoring: the term ID is the term itself, so it cannot be changed");
    }

    @Override
    public String getID() {
        return name;
    }
}
